src/librustc_data_structures/bitvec.rs

```diff
@@ -293,15 +293,23 @@ impl<R: Idx, C: Idx> SparseBitMatrix<R, C> {
     ///
     /// Returns true if this changed the matrix, and false otherwise.
     pub fn add(&mut self, row: R, column: C) -> bool {
-        self.vector[row].insert(column)
+        if let None = self.vector.get(row) {
+            self.vector.push(SparseBitSet::new());
```

Comment:
This doesn't seem right -- imagine that the vector starts out as empty, but we invoke `add(2, 0)` -- then we have to push two things. I think the method you want is either [`resize`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize) or -- perhaps mildly more efficient -- [`resize_with`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize_with). If `index` where a plain vector, you would do something like this:

```rust
self.vector.resize_with(row + 1, || SparseBitSet::new())
```

However, that won't work because `vector` is an `IndexVec` and it doesn't offer `resize_with`. You could use `self.vector.raw.resize_with`, but better would be to add a method to `IndexVec`:

```rust
impl<I, T> IndexVec<I, T> {
    fn resize_to_elem(&mut self, elem: I, fill_value: impl FnMut() -> T) {
        let min_new_len = elem.as_usize() + 1;
        self.raw.resize_with(min_new_len, fill_value);
    }
}
```

and then invoke `self.vector.resize_to_elem(row, || SparseBitSet::new());`.

Reply:
Fixed this.